cub/cub/device/dispatch/tuning/tuning_run_length_encode.cuh

```diff
@@ -431,6 +513,86 @@ struct sm90_tuning<LengthT, __uint128_t, primitive_length::yes, primitive_key::n
+{
+  // ipt_20.tpb_224.trp_1.ts_0.ld_1.ns_64.dcid_2.l2w_315 1.119878  1.003690  1.130067  1.338983
+  static constexpr int threads                       = 224;
+  static constexpr int items                         = 20;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr bool store_with_time_slicing      = false;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_CA;
+  using delay_constructor                            = detail::exponential_backoff_constructor_t<64, 315>;
+};
+
+template <class LengthT, class KeyT>
+struct sm100_tuning<LengthT, KeyT, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_2>
+{
+  // ipt_20.tpb_224.trp_1.ts_0.ld_0.ns_116.dcid_7.l2w_340 1.146528  1.072769  1.152390  1.333333
+  static constexpr int threads                       = 224;
+  static constexpr int items                         = 20;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr bool store_with_time_slicing      = false;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::exponential_backon_constructor_t<116, 340>;
+};
+
+template <class LengthT, class KeyT>
+struct sm100_tuning<LengthT, KeyT, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_4>
+{
+  // ipt_13.tpb_224.trp_0.ts_0.ld_0.ns_252.dcid_2.l2w_470 1.113202  1.003690  1.133114  1.349296
+  static constexpr int threads                       = 224;
+  static constexpr int items                         = 13;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_DIRECT;
+  static constexpr bool store_with_time_slicing      = false;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::exponential_backoff_constructor_t<252, 470>;
+};
+
+template <class LengthT, class KeyT>
+struct sm100_tuning<LengthT, KeyT, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_8>
+{
+  // ipt_15.tpb_256.trp_1.ts_0.ld_0.ns_28.dcid_2.l2w_520 1.114944  1.033189  1.122360  1.252083
+  static constexpr int threads                       = 256;
+  static constexpr int items                         = 15;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr bool store_with_time_slicing      = false;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::exponential_backoff_constructor_t<28, 520>;
+};
+
+// need to default it back explicitly because it regresses
+template <class LengthT>
+struct sm100_tuning<LengthT, double, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_8>
+    : sm90_tuning<LengthT, double, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_8>
+{};
```

Comment:
I want to figure out why this happens.

Reply:
The SASS diff is large when I comment this one out, so something is up.

Reply:
OK, I got it. This partial template specialization makes the previous tuning
```
template <class LengthT, class KeyT>
struct sm100_tuning<LengthT, KeyT, primitive_length::yes, primitive_key::yes, length_size::_4, key_size::_8>
```
fall back to Policy900 if `KeyT` is `double`.

Reply:
Updated the comment.